hawkbit-dmf/hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/AmqpFormulateErrorMessage.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2021 Bosch.IO GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.amqp;
+
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.rabbit.support.ListenerExecutionFailedException;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Class that formulates a meaningful error message
+ */
+public class AmqpFormulateErrorMessage {
+
+    /**
+     * Constructs an error message based on failed message content
+     *
+     * @param throwable
+     *                  the throwable containing failed message content
+     * @return
+     *                  meaningful error message
+     */
+    public String constructErrorMessage(final Throwable throwable) {
+        StringBuilder completeErrorMessage = new StringBuilder();
+
+        //extract failed message content
+        Collection<Message> failedMessages = ((ListenerExecutionFailedException) throwable).getFailedMessages();
```

Comment:
you should first check if `throwable` is an instance of `ListenerExecutionFailedException` and if it is not the case just return `throwable.getMessage()` probably combined with `throwable.getCause().getMessage()`, otherwise you would extract the properties/headers from the `ListenerExecutionFailedException`

Reply:
throwable.getMessage() and throwable.getCause().getMessage() both would be containing the same message hence will return throwable.getCause().getMessage()